Project Management Process.md

```diff
@@ -0,0 +1,287 @@
+    Milestone A     |     Milestone 2           Goal C
+    Milestone B     |       Goal 4              Goal D
+    Milestone C     |     Milestone 3      
+                    |       Goal 5      
+  Projects          |  
+    Project 1       |   Backlog             Backlog
+    Project 2       |     Goal 1              Goal A
+    Project 3       |     Goal 2              Goal B
+                    |     Goal 3              Goal C
+                    |     Goal 4              Goal D
+                    |     Goal 5
+```
+
+### Goals
+At the heart of each project are the goals. They are the basic unit of work. Goals can be anything that ***adds value to a release***. This includes issues reported by users, feature requests, bugs, Pull-Requests, general tasks such as refactoring, documentation or research. Goals are collected, maintained and tracked in the backlog by the project lead.
+
+A goal description includes the following:
+- Name / Summary
+  - Describes what the goal is on high level
+  - Example 1: "Reduce the size of the go-ipfs executable"
+  - Example 2: "js-ipfs should have API Documentation"
+- Description
+  - What the problem is and how to achieve this goal
+  - Should tell you how you will know that the problem has been solved or the goal has been achieved
+  - Describe the type of this goal: is it a bug? a new feature? something else?
+- Status
+  - The status should signal the goal's status towards completion
+  - What is the current state of this goal? Is it to-be-reviewed? In progress? Blocked? Done?
+  - A goal can be in only one state at a time
+- Dependencies (where applicable)
+  - Links to other goals that must happen before or concurrently with this one
+- Any additional information that helps to reach this goal (optional)
+
+TODO: a clear example
+
+### Backlog
+Backlog is the collection of all goals in a project. It should be as comprehensive as possible, including everything that needs to happen or should happen in a particular project. The backlog indicates priority by order: the items that have the highest priority are at the top of the list. The backlog is owned and maintained by the project lead. As part of daily work, the backlog gets updated and a general grooming should happen on weekly basis.
+
+A backlog has the following information:
+- All goals of a project
+- Priority of goals in sorted order
+
+While the backlog is usually one big list of goals, it may sometimes become convoluted. In this case, the backlog list can be split in two: ***Backlog*** and ***Future Work***. The future work section lists goals that will probably not be worked on anytime soon, or are not ready to be worked on yet, whereas the ready section contains the goals that are ready to be worked on or "approved". If split, together they are called "the backlog" while *Future Work* is referenced only in its specific meaning. The split and maintenance of both lists are done by the project lead.
```

Comment:
Regarding @diasdavid's comment 
> One thing that I feel missing, is a way to communicate if something is in our roadmap, but just not right now. e.g: I've been deferring all the examples + tutorial issues with the label Milestone 4, this way I can show our intent to handle it during the expected timeline defined by the roadmap, but also not make it distracting by being on the backlog, nor in the Future Work area, which might sound it was forgotten.

I think that is the purpose of Milestones -- to show that certain work is on the roadmap, and where it is on that roadmap.  By contrast, the backlog and the Icebox/Unstaged will both contain lots of stuff that's listed in Milestones.  The milestones show you the Roadmap view so you can answer questions like
* What is our trajectory? 
* What are our priorities?
* When are things likely to be done?

Meanwhile the pipeline with Icebox, Ready, In Progress, etc. gives you the working view so you can answer questions like
* What's being worked on _now_? 
* What's ready to be worked on _now_? 
* What's been completed recently? 

Given that, could you clarify what you think is missing? Are you saying that the _Roadmap_ should have milestones that are marked "Future Work", indicating that they're things we haven't explicitly scheduled but we have them on the long-term Roadmap?